commands.js

```diff
@@ -5,8 +5,229 @@
  * to run. A lot of these are sent by the client.
  *
- * If you'd like to modify commands, please go to config/commands.js,
- * which also teaches you how to use commands.
+ * If you'd like to modify commands, please go to the chat-plugins
+ * folder.
+ *
+ * You can define the commands such as 'whois', then use it by
+ * typing /whois into Pokemon Showdown.
+ *
+ * A command can be in the form:
+ *   ip: 'whois',
+ * This is called an alias: it makes it so /ip does the same thing as
+ * /whois.
+ *
+ * But to actually define a command, it's a function:
+ *
+ *   avatars: function (target, room, user) {
+ *     if (!this.canBroadcast()) return;
+ *     this.sendReplyBox('You can <button name="avatars">change ' +
+ *       'your avatar</button> by clicking on it in the <button ' +
+ *       'name="openOptions"><i class="icon-cog"></i> Options' +
+ *       '</button> menu in the upper right. Custom avatars are ' +
+ *       'only obtainable by staff.');
+ *   }
+ *
+ * Commands are actually passed five parameters:
+ *   function (target, room, user, connection, cmd, message)
+ * Most of the time, you only need the first three, though.
+ *
+ * target = the part of the message after the command
+ * room = the room object the message was sent to
+ *   The room name is room.id
+ * user = the user object that sent the message
+ *   The user's name is user.name
+ * connection = the connection that the message was sent from
+ * cmd = the name of the command
+ * message = the entire message sent by the user
+ *
+ * If a user types in "/msg zarel, hello"
+ *   target = "zarel, hello"
+ *   cmd = "msg"
+ *   message = "/msg zarel, hello"
+ *
+ * Commands return the message the user should say. If they don't
+ * return anything or return something falsy, the user won't say
+ * anything.
+ *
+ * A command can also be an object, in which case is treated like
+ * a namespace:
+ *
+ *   game: {
+ *     play: function (target, room, user) {
+ *       user.isPlaying = true;
+ *       this.sendReply("Playing.");
+ *     },
+ *     stop: function (target, room, user) {
+ *       user.isPlaying = false;
+ *       this.sendReply("Stopped.");
+ *     }
+ *   }
+ *
+ * These commands can be called by '/game play' and '/game stop'.
+ * Namespaces help organise commands, and nest them under
+ * one main command.
+ * Note: Multiple namespaces can be nested, but the final (innermost)
+ *       command must be a function.
+ *
+ * Namespace objects can have help entries and so can the eternal
+ * commands:
+ *
+ *   game: {
+ *     play: function (target, room, user) {
+ *       user.isPlaying = true;
+ *       this.sendReply("Playing.");
+ *     },
+ *     playhelp: ["Tells you if the user is playing."]
+ *   },
+ *   gamehelp: ["commands for /game are:",
+ *     "/game play - Tells you if the user is playing.",
+ *     "/game stop - Tells you if the user stopped playing."]
+ *
+ * The help entries are accessed with "/help game play" and
+ * "/help game" respectively.
```

Comment:
Should I consolidate this with the previous example, invent a new namespace function as an example, or is this adequate at explaining how to use help entries with namesapce?